fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryObject.java

```diff
@@ -74,21 +74,29 @@ public URI getUri() {
 	 * @throws FedoraException
 	 */
 	public Model getModel() throws FedoraException {
-		if (model == null) {
-			dataLoader.loadModel(this);
-		}
-
+		dataLoader.loadModel(this);
 		return model;
 	}
 
 	/**
+	 * Return true if the model for this object has been populated
+	 * 
+	 * @return
+	 */
+	public boolean hasModel() {
+		return model != null;
+	}
+
+	/**
 	 * Set the model
 	 * 
 	 * @param model
 	 * @return
 	 */
 	public void setModel(Model model) {
 		this.model = model;
+		// Clear the cached types list
```

Comment:
The list of types needs to be cleared whenever the model is changed to avoid becoming stale since it is just a cached subset of values from the model.  I could shift that logic to the data loader though if that'd make more sense